src/test/java/tests/CollaboratorServiceExTests.java

```diff
@@ -0,0 +1,62 @@
+package tests;
+
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import github.CollaboratorServiceEx;
+import org.eclipse.egit.github.core.RepositoryId;
+import org.eclipse.egit.github.core.User;
+import org.eclipse.egit.github.core.client.GitHubClient;
+import org.junit.Test;
+import org.mockserver.client.server.MockServerClient;
+import org.mockserver.integration.ClientAndServer;
+
+import java.io.IOException;
+import java.lang.reflect.Type;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockserver.model.HttpRequest.request;
+import static org.mockserver.model.HttpResponse.response;
+
+public class CollaboratorServiceExTests {
+
+    @Test(expected = IllegalArgumentException.class)
+    public void getCollaborators_invalidRepoId() throws IOException {
+        GitHubClient client = new GitHubClient();
+        CollaboratorServiceEx service = new CollaboratorServiceEx(client);
+
+        service.getCollaborators(null);
+        service.getCollaborators(RepositoryId.createFromId(""));
+        service.getCollaborators(RepositoryId.createFromId("fakeowner/bogusrepo"));
+    }
+    
+    @Test
+    public void getCollaborators_validRepoId_successful() throws IOException {
+        MockServerClient mockServer = ClientAndServer.startClientAndServer(8888);
+        String sampleCollaborators = TestUtils.readFileFromResource(this, "tests/CollaboratorsSample.json");
+
+        mockServer.when(
+                request()
+                    .withPath(TestUtils.API_PREFIX + "/repos/hubturbo/tests/collaborators")
+        ).respond(
+                response()
+                    .withBody(sampleCollaborators)
+        );
+
+        GitHubClient client = new GitHubClient("localhost", 8888, "http");
+        CollaboratorServiceEx service = new CollaboratorServiceEx(client);
+
+        Type listOfUsers = new TypeToken<List<User>>() {}.getType();
+        List<User> expectedCollaborators = new Gson().fromJson(sampleCollaborators, listOfUsers);
+        List<User> actualCollaborators = service.getCollaborators(RepositoryId.createFromId("hubturbo/tests"));
+
+        assertEquals(expectedCollaborators.size(), actualCollaborators.size());
+
+        for (int i = 0; i < expectedCollaborators.size(); i++) {
+            assertEquals(expectedCollaborators.get(i).getLogin(), actualCollaborators.get(i).getLogin());
+            assertEquals(expectedCollaborators.get(i).getName(), actualCollaborators.get(i).getName());
```

Comment:
Maybe we could check here that the names of the actual collaborators aren't empty.